docs/deployments.md

```diff
@@ -1,306 +1,172 @@
+
+Each time a new deployment is created, the `latestVersion` field of `deploymentConfig` is incremented, and a `deploymentCause` is added to the `deploymentConfig` describing the change that led to the latest deployment.
+
+## Defining a deploymentConfig
+
+A `deploymentConfig` in OpenShift is a REST object which can be POSTed to the API server to create a new instance. Consider a simple configuration which should result in a new `deployment` every time a Docker image tag changes.
+
+```
+{
+  "metadata": {
+    "name": "frontend"
+  },
+  "kind": "DeploymentConfig",
+  "apiVersion": "v1beta1",
+  "triggers": [
+    {
+      "type": "ImageChange",
+      "imageChangeParams": {
+        "automatic": true,
+        "containerNames": [
+          "helloworld"
+        ],
+        "repositoryName": "openshift/origin-ruby-sample",
+        "tag": "latest"
+      }
+    }
+  ],
+  "template": {
+    "strategy": {
+      "type": "Recreate"
+    },
+    "controllerTemplate": {
+      "replicas": 1,
+      "replicaSelector": {
+        "name": "frontend"
+      },
+      "podTemplate": {
+        "desiredState": {
+          "manifest": {
+            "version": "v1beta1",
+            "containers": [
+              {
+                "name": "helloworld",
+                "image": "openshift/openshift/origin-ruby-sample",
+                "ports": [
+                  {
+                    "containerPort": 8080
+                  }
+                ]
+              }
+            ]
+          }
+        },
+        "labels": {
+          "name": "frontend"
+        }
+      }
+    }
+  }
+}
+```
+
+This specification will create a new `deploymentConfig` named `frontend`. A single ImageChange `trigger` is defined, which causes a new `deployment` to be created each time the `openshift/origin-ruby-sample:latest` tag value changes. The Recreate `strategy` makes the `deployment` live by removing any prior `deployment` and increasing the replica count of the new `deployment`.
+
+## Triggers
+
+A `deploymentConfig` contains `triggers` which drive the creation of new deployments in response to events (both inside and outside OpenShift). The following trigger types are supported:
+
+##### Image change triggers
+
+The ImageChange `trigger` will result in a new deployment whenever the value of a Docker `imageRepository` tag value changes. Consider an example trigger.
+
+```
+{
+  "type": "ImageChange",
+  "imageChangeParams": {
+    "automatic": true,
+    "containerNames": [
+      "helloworld"
+    ],
+    "repositoryName": "openshift/origin-ruby-sample",
+    "tag": "latest"
+  }
+}
+```
+
+In this example, when the `latest` tag value for the `imageRepository` named `openshift/origin-ruby-sample` changes, every container in the `deploymentConfig` matching those in `containerNames` will be updated with the new tag value, and a new `deployment` will be created.
```

Comment:
"every container in the `deploymentConfig` matching those in `containerNames`" reads funny to me.  

maybe: the containers specified in `containerNames` for `deploymentConfig` D will be updated ...